.github/workflows/end2end.yml

```diff
@@ -32,28 +32,28 @@ jobs:
           GH_TOKEN: ${{ github.token }}
           PR_NUMBER: ${{ github.event.issue.number }}
 
-      - name: Install Foundry
-        uses: foundry-rs/foundry-toolchain@v1
-        with:
-          version: nightly
+      #- name: Install Foundry
+      #  uses: foundry-rs/foundry-toolchain@v1
+      #  with:
+      #    version: nightly
 
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.8
+      #- name: Set up Python
+      #  uses: actions/setup-python@v4
+      #  with:
+      #    python-version: 3.8
 
-      - name: Deploy main contract
-        run: |
-          echo "CONTRACT_ADDRESS=$(forge script script/Deployment.s.sol:NovaVerifierDeployer --fork-url $ANVIL_URL --private-key $ANVIL_PRIVATE_KEY --broadcast --non-interactive | sed -n 's/.*Contract Address: //p' | tail -1)" >> $GITHUB_OUTPUT
-        id: deployment
+      #- name: Deploy main contract
+      #  run: |
+      #    echo "CONTRACT_ADDRESS=$(forge script script/Deployment.s.sol:NovaVerifierDeployer --fork-url $ANVIL_URL --private-key $ANVIL_PRIVATE_KEY --broadcast --non-interactive | sed -n 's/.*Contract Address: //p' | tail -1)" >> $GITHUB_OUTPUT
+      #  id: deployment
 
-      - name: Load proof and public parameters
-        run: |
-          python loader.py pp-verifier-key.json pp-compressed-snark.json ${{steps.deployment.outputs.CONTRACT_ADDRESS}} $ANVIL_URL $ANVIL_PRIVATE_KEY
+      #- name: Load proof and public parameters
+      #  run: |
+      #    python loader.py pp-verifier-key.json pp-compressed-snark.json ${{steps.deployment.outputs.CONTRACT_ADDRESS}} $ANVIL_URL $ANVIL_PRIVATE_KEY
 
-      - name: Check proof verification status
-        run: |
-          [[ $(cast call ${{steps.deployment.outputs.CONTRACT_ADDRESS}} "verify(uint32,uint256[],uint256[],bool)(bool)" "3" "[1]" "[0]" "true" --private-key $ANVIL_PRIVATE_KEY --rpc-url $ANVIL_URL) == true ]] && exit 0 || exit 1
+      #- name: Check proof verification status
+      #  run: |
+      #    [[ $(cast call ${{steps.deployment.outputs.CONTRACT_ADDRESS}} "verify(uint32,uint256[],uint256[],bool)(bool)" "3" "[1]" "[0]" "true" --private-key $ANVIL_PRIVATE_KEY --rpc-url $ANVIL_URL) == true ]] && exit 0 || exit 1
 
```

Comment:
Did you mean to keep it commented? We could have some conditions to make it run when `!test` is commented, but only on `feature/*` branches?

Also, should we have this flow possible on both issues and PRs or only PRs?

Reply:
For the `main` branch we don't need integration testing - though it is relevant for other branches. Let's involve @samuelburnham to provide some input on this - as he initially helped with that stuff. 

> We could have some conditions to make it run when `!test` is commented, but only on feature/* branches

That seems to be quite good approach

Reply:
I can tackle running on `feature/*` branches, sounds like a great plan.

> Also, should we have this flow possible on both issues and PRs or only PRs?

PRs only I think, if we want to run on branches without an open PR it can be done via [`workflow_dispatch`](https://github.com/lurk-lab/solidity-verifier/actions/workflows/ci.yml) from https://github.com/lurk-lab/solidity-verifier/blob/main/.github/workflows/ci.yml#L8